spec/models/playlist_spec.rb

```diff
@@ -101,17 +101,17 @@
   context "favorites" do
     it 'should create a new playlist for a user who does not have one' do
       expect(users(:sandbags).playlists.favorites).not_to be_present
-      users(:sandbags).toggle_favorite(assets(:valid_mp3))
+      users(:sandbags).tracks.favorites.scoping { Track.create(asset_id: assets(:valid_mp3).id) }
```

Comment:
Take care of deprecation warning
```
DEPRECATION WARNING: Class level methods will no longer inherit scoping from `create` in Rails 6.1. To continue using the scoped relation, pass it into the block directly. To instead access the full set of models, as Rails 6.1 will, use `Track.unscoped`.
```

Took idea for this fix from https://github.com/stefankroes/ancestry/pull/442#issuecomment-530160300

Reply:
The desired solution is `To instead access the full set of models, as Rails 6.1 will, use Track.unscoped`. I've tried looking through the rails issues and PRs, tried changing scope - it fixed deprecation warning but broke all tests. 

Suggestion on what this message mean and how I can fix it?

Reply:
It looks to me they want to explicitly disallowing "reaching through" an association to do something like .create/.new/.where if you have scopes on the first model.

Meaning `Track.create` is fine but `Track.favorites.create` has a scope before the create, which is confusing and possibly problematic (apparently).

In this case, `where(is_favorite: true)` is useful for looking up records, but isn't going to do what we might want it to do (set `is_favorite` to `true`) since it's a `where` clause for a select query. 

I'd go for the straightforward thing you want to do (can also throw out the .id calls)

```suggestion
      users(:sandbags).tracks.create(asset: assets(:valid_mp3, is_favorite: true))
```

So cahdec

Reply:
Well, you see? Your words make sense... why can't they say that :D